Global refactoring

T1xx1 opened the pull request:
Issue: #27.

That's the README after the key-value pairs swap.

I'll make this pr a draft for two reasons.
- I haven't changed the introduction yet as discussed.
- There are many points, that I'm listing down here, that I want to discuss with you before proceeding.

So don't merge until the review.

Also, search for errors because I'm not sure that everything is fine.

Get comfy because it's a looong reading.

---

## Changes

### Plur contexts
I think it will be better to have plur contexts than in the single form.
E.g. {binary operands} instead of {binary operand}
Just for style.

### E.g.
E.g. finally got the discussed "for example" as word instead of "example" and has been moved to the f section.

---

## Ambiguous

### Dev
- development, developer and device.

Development and developer might have "dev" as they are similar but "dev" of device should be 🔴. 

### Def
- definition, default and define.

Like "dev" definition and define with 🟢 "def" and default with 🔴.

### Temp
- temperature and temporary.

I suggest "temp" for temporary as the windows os temporary file folder and "T" for temperature like in physics formulas.

---

## Conflicts

### Argument
- arg can also be used in functions so why has the {memory} context?
- argument is 🟡 but parameter is 🟢.

Here I don't know what to do.

### V section is a mess
This section made me go crazy. I found 6 duplicates (that I removed) and there are 2 conflicts with the same abbr.

![Screenshot (86)](https://user-images.githubusercontent.com/69322987/215205104-7558a3f9-8c54-4ae5-98cc-f5603656c18c.png)

So that's my plan
- "Ver" is used for vertical so "ver" of version goes 🔴
- Version gets "v" 🟢.
- Value gets "v" 🔴 because version already has it and because I also remember that we discussed it before and concluded that's not a good one
- "val" of value stays 🟢.
- And finally vector is yellow but missing the context so I'll go with 🟡 {physic}.

After the changes.

```
- value
	- 🟢 val
	- 🔴 v
- vector • 🟡 v {physic}
- variable • 🟢 var
- version
	- 🟢 v
	- 🔴 ver
- vertical • 🟢 ver
```

No conflicts 😀

### User
- user don't have a 🟢 abbr

Idk.

### Short abbrs

![Screenshot (87)](https://user-images.githubusercontent.com/69322987/215205449-81089610-966c-4cbf-8fcc-a74e8d45b4a5.png)

However all along the list, I've found many abbr with 🟢 that are 5 letters long and are abbreviated in 3 letters like:

![Screenshot (88)](https://user-images.githubusercontent.com/69322987/215205467-56026e9f-5806-4ee5-be42-f04be9cf9fb2.png)
![Screenshot (89)](https://user-images.githubusercontent.com/69322987/215205493-b6a6d0c8-8717-4d45-8ceb-01975429ff12.png)

I can close an eye on "img" because it's obvious.
The real problem comes where there are other abbrs with the same length marked as red (text and copy).
We're missing cohesion here with the rules and also with the list itself.

### Red with context
Something is clearly wrong here. I found 2 abbrs having 🔴 but with a context after. Is the degree or the context wrong?

![Screenshot (90)](https://user-images.githubusercontent.com/69322987/215205557-8c98955f-aba8-4869-9543-2b803b152b40.png)

![Screenshot (91)](https://user-images.githubusercontent.com/69322987/215205552-3e9465e5-4956-4a10-b514-b81ea21304fd.png)

### Yes and no
Talking about the no abbr.
- No is 🔴 but yes is 🟡 .

No is wrong as sometimes n and y are used in terminals and CLIs.

### Loops
- Increase, decrease and iterator are 🟢.

I don't see them 🟢 as they are only used in loops contexts. I suggest 🟡 {loops}

---

## New abbrs
Many new abbrs came to my mind reading the list.
- octal • oct (like hex and dec)
- controller • ? (we have microcontroller so why not controller)
- typescript • ts (we have js)
- string length • strlen (python and jupiter)
- initialization • init (we have initialize)
- corporation • corp
- organization • org
- standard input output • stdio (we have input and output)
- description • desc (we have descriptor)

Okay, I'm done.
Looking for your responses. Good night.

philipeachille commented:
Nice work!! Here are my comment, suggestions and opinions

"Plur contexts" -> fine with me
"E.g." -> fine with me
"dev" -> development 🟢, developer 🟢, device 🟡
"def" -> agree
"temp" -> agree
"arg" -> argument 🟢
"v" -> I'd define this section like this:
- value
	- 🟢 val
	- 🟡 v { with k } _[ btw I like whitespace after and before brackets ]_
- vector • 🟡 v { physics }
- variable • 🟢 var
- version
	- 🟢 v
	- 🔴 ver
- vertical • 🟢 ver

"user" -> is short already, no abbrev needed?
"Short abbrs" -> we should not be so hard on the rules. sometimes it's just fine
"ord" -> order 🟡
"n" -> no 🟡 { with y }
"y" -> yes 🟡 { with n }
loops -> agree
New abbrs -> add individual issues after refactor?

T1xx1 commented:
Great. I'll wail @kisvegabor and, if I can, get everything done by tomorrow.

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
+-  column • 🟢 col
+-  concatenate • 🟢 concat
+-  configuration
+   -  🟢 conf
+   -  🟢 config
```
There should be on one green per word. I vote for "config".

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
+-  cotangent • 🟢 cot
+-  counter • 🟢 cnt
+-  current
+   -  🟢 cur
+   -  🟢 curr
```
Same here, but I don't know which one is better.

T1xx1 replied:
As they don't different much from each other I suggest "curr"

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
+-  for example 🟢 e.g.
+-  format • 🟢 fmt
+-  function
+   -  🟢 f
+   -  🟢 fn
+   -  🟢 func
```
I'd keep "func".

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
+-  integer • 🟢 int
+-  iterator • 🟢 iter
+-  interface
+   -  🟢 iface
+   -  🟢 intf
```
IDK which is better, needs some research.

T1xx1 replied:
Idk too

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
+-  picture • 🟢 pic
+-  pixel • 🟢 px
+-  pointer
+   -  🟡 p {memory}
+   -  🟢 ptr
```
There are only a few places, where more than one option is added. Does it really wort breaking the format?

What about this:
pointer •  🟡 p {memory} • 🟢 ptr

T1xx1 replied:
I think I will become hard to read.

kisvegabor replied:
IMO all inline would nice and unified, and we should use it if possible. Can you imagine other formats other ideas for inlining?

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
+-  rectangle • 🟡 rect {shape}
+-  reference • 🟢 ref
+-  regular expression
+   -  🟢 rgx
+   -  🟢 regex
```
"regex" is much better IMO.

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
--  🔴 txt • text
+-  temperature • 🟢 temp
+-  temporary
+   - 🟢 tmp
+   - 🟢 temp
```
If "temp" is for "temperature" we should use "tmp" here.

T1xx1 replied:
I prefer "temp" for temporary and "T" for temperature as I said in the first comment.

kisvegabor replied:
My only issue with "T" is that if naming conventions requires "all_lower_case" format then e.g. "living_room_t" will be hard to decode. Anyway, let's pick one, to finalize just the format and we can think about it the next iteration.

kisvegabor commented on README.md:
```diff
@@ -44,278 +44,316 @@ Unrecommended ones can still be considered. The most important thing is to remai
+   -  🟡 u {URLs}
+   -  🔴 usr
 
 ### V
```
I mostly agree with the suggestions of @philipeachille regarding this section.

kisvegabor reviewed (commented):
Hi, 

great work! 

I think we shouldn't bother too much with the abbreviations themselves, but focus on the format now, and open new PR to review the existing suggestions.

So my comments are mostly about the format of the list and the rule breaking parts (e.g. there should be only 1 green per word).

kisvegabor reviewed (commented):
Great, it seems we agree on most of the points. :slightly_smiling_face:

T1xx1 commented:
> Great, it seems we agree on most of the points. 🙂

OK. We will discuss on "T" later.

I can try thinking about other formats.

T1xx1 commented:
Introduction refactor
- I removed the index [Discussion](https://github.com/kisvegabor/abbreviations-in-code/issues/24#issuecomment-1385947503)
- I changed the description with the "Goal" paragraph, it suits a lot in this role.
- I decided to move the guide instead of reducing it or deleting it. It has a new file so someone new can view it. However, someone who already knows the rules can scroll to the list and it doesn't take up space in the README.

I readded the symbols section. (I've deleted it by mistake previously).
I added the other section just in case we aren't sure where to put an abbr while we discuss it.
I added the license link at the bottom.

Now I'm adding all the points we have discussed.

Another thing. To refactor the CONTRIBUTING I need to create the ISSUE_TEMPLATES to separate how to contribute from how to insert things in the list. Can I proceed?

kisvegabor commented:
Thank you Tiziniao!

Why do we need bot guide.md and CONTRIBUTING.md? It seems redundant to me. 

> Another thing. To refactor the CONTRIBUTING I need to create the ISSUE_TEMPLATES to separate how to contribute from how to insert things in the list. Can I proceed?

Sure!

T1xx1 commented:
> Thank you Tiziniao!

You're welcome, HAHAH BTW my name is Tiziano.


> Why do we need bot guide.md and CONTRIBUTING.md? It seems redundant to me. 

The guide is the previous step before the
CONTRIBUTING. 
The guide shows how to use the list while the contributing shows how people can help adding new abbrs, obviously following the rules in the guide.
Before adding new abbrs you need to know how to use the list so you read the guide and after use the contributing file (and I'm adding the links soon) to open new issues with the ISSUE_TEPLATES.
That's why I asked for those.


> > Another thing. To refactor the CONTRIBUTING I need to create the ISSUE_TEMPLATES to separate how to contribute from how to insert things in the list. Can I proceed?
> 
> Sure!

Great.
Meanwhile we wait for @philipeachille response I'm adding the points we discussed.

kisvegabor commented:
>  HAHAH BTW my name is Tiziano.

Ahhhh, I promised to myself that I will always copy paste names, to be sure I won't mistype them. I tried typing it myself once and see what happened. Sorry :sweat_smile: 

> The guide shows how to use the list while the contributing shows how people can help adding new abbrs, obviously following the rules in the guide.

I'm not sure people will click and read the guide. IMO the we can add something like this into the README:

### How to use this list?
-  🟢 Recommended: clear in all context
-  🟡 Context-sensitive: clear only in a given context. The context is specified between `{}` after the abbreviation
-  🔴 Not recommended: common, but not recommended abbreviations.

---

The alphabetic order and other things shouldn't be required to use the list.

T1xx1 commented:
> > HAHAH BTW my name is Tiziano.
> 
> Ahhhh, I promised to myself that I will always copy paste names, to be sure I won't mistype them. I tried typing it myself once and see what happened. Sorry 😅

HAHAH don't worry.

> 
> > The guide shows how to use the list while the contributing shows how people can help adding new abbrs, obviously following the rules in the guide.
> 
> I'm not sure people will click and read the guide.

We should not force them to see it if they don't want.

> IMO the we can add something like this into the README:

I think it should be the clearest as possible, without anything that is not the list.

> The alphabetic order and other things shouldn't be required to use the list.

Yeah but these are still things you should be aware of to read the list.

But most of all we need them to state rules for contributions and edits. As new abbrs have to follow the order specified in the guide.

kisvegabor commented on guide.md:
```diff
@@ -0,0 +1,18 @@
+
+# Guide
+
+1. The list is sorted with alphabetical sections. With exceptions for "Numbers", "Symbols" and "Others" which are special.
+2. In section, abbrs are sorted in alphabetical order.
+3. Abbrs are in word-abbr format.
```
These should be obvious at first look and we don't need to state these directly.

T1xx1 replied:
Only wanted to be clear but ok.

kisvegabor commented on guide.md:
```diff
@@ -0,0 +1,18 @@
+Unrecommended abbrs can still be considered.
+The most important thing is to remain consistent in your code.
+
+4. Context abbrs are more suitable in certain cases. The context is specified between { } after the abbr.
```
We can add this next to :yellow_circle:

T1xx1 replied:
In the contributing?

kisvegabor replied:
I've still followed my proposal: we need to add only :green_circle:  :yellow_circle:  and :red_circle:  in the README, with one sentence explanation for each. And we can explain there what `{ }` means in case of :yellow_circle:

kisvegabor commented on guide.md:
```diff
@@ -0,0 +1,18 @@
+The most important thing is to remain consistent in your code.
+
+4. Context abbrs are more suitable in certain cases. The context is specified between { } after the abbr.
+5. If a word has more possible abbrs they are sorted from the most recommended to the least one. If there's more of one kind it's sorted in alphabetical order.
```
Each word should have only one recommended abbreviation.

T1xx1 replied:
Okk.

philipeachille commented:
"Meanwhile we wait for @philipeachille response" @T1xx1 what do you wish me to comment on?

kisvegabor commented:
> We should not force them to see it if they don't want.
> I think it should be the clearest as possible, without anything that is not the list.
> Yeah but these are still things you should be aware of to read the list.

I've added comments to guide.md. For me it seems only the colored circles needs to be explained and for this it's not worth to have separate file.

> But most of all we need them to state rules for contributions and edits. As new abbrs have to follow the order specified in the guide.

These should be stated in CONTRIBUTING.md. We shouldn't bother the user with rules and things that are required only for the contributors.

T1xx1 commented:
> "Meanwhile we wait for @philipeachille response" @T1xx1 what do you wish me to comment on?

Just to see if you agree with the changes.

T1xx1 commented:
How does it look now?

kisvegabor commented on README.md:
```diff
@@ -1,320 +1,353 @@
+-  🔴 Not recommended
 
-Unrecommended ones can still be considered. The most important thing is to remain consistent in your code.
+Unrecommended abbrs can still be considered.
```
Can they? Isn't the whole point of this list to not consider un-recommended abbreviations but stick with the recommended ones?

T1xx1 replied:
When I did this commit, I was late, so I just sticked all the text without making changes.

kisvegabor replied:
Okay :slightly_smiling_face: 
I' think the best would be to remove this statement.

kisvegabor commented on README.md:
```diff
@@ -1,320 +1,353 @@
 
-Unrecommended ones can still be considered. The most important thing is to remain consistent in your code.
+Unrecommended abbrs can still be considered.
+The most important thing is to remain consistent in your code.
```
Do we really need to state this?

kisvegabor commented on README.md:
```diff
@@ -1,320 +1,353 @@
-Unrecommended ones can still be considered. The most important thing is to remain consistent in your code.
+Unrecommended abbrs can still be considered.
+The most important thing is to remain consistent in your code.
+Context abbrs are more suitable in certain cases. The context is specified between { } after the abbr.
```
Can we update :yellow_circle:  as
-  🟡 Context-sensitive (You can see the context in `{ }` after the abbreviation)

T1xx1 replied:
yes

kisvegabor commented on README.md:
```diff
@@ -1,320 +1,353 @@
+-  record • 🟢 rec
+-  rectangle • 🟡 rect { shapes }
+-  reference • 🟢 ref
+-  regular expression
+   -  🟢 regex
+   -  🔴 rgx
```
Can we still figure out something for this kind of cases?

I think it doesn't look that good if have only a few lines that break the nice pattern. I still vote for inlining. The worst case I can image in the practice is 5 abbrevs: 1 green, 2 yellow, 2 red.

So:
word • 🟢 w1 • 🟡 w2 { context } • 🟡 w3 { context } • 🔴 w4 • 🔴 w4

Or to stick with this specific case:
regular expression • 🟢 regex • 🔴 rgx

T1xx1 replied:
we can wrap it in inline when there are maximum 3 abbrs.

kisvegabor replied:
Now we have 3 abbrs in one place and 2 abbrs on a few places.  So max 3 works now, and we can figure it out later when there will be more abbrs somewhere.

kisvegabor commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,12 @@
+-  `significant` • seen for the first time you still know what it means
+
+## Open an issue
+- [New abbr](./ISSUE_TEMPLATE/newAbbr.MD)
+- [Edit curr abbr](./ISSUE_TEMPLATE/editCurrAbbr.md)
```
Shall we really need to abbreviate here? :smile: 

```suggestion
- [New abbreviation](./ISSUE_TEMPLATE/newAbbr.MD)
- [Edit current abbreviation](./ISSUE_TEMPLATE/editCurrAbbr.md)
```

T1xx1 replied:
thought it was funny to use the abbr we state for codes

kisvegabor replied:
Yeah, I see you point and it's really funny :slightly_smiling_face: 

Ok, let's keep your suggestion!

T1xx1 replied:
HAHA ok👌

kisvegabor commented on .github/ISSUE_TEMPLATE/editCurrAbbr.md:
```diff
@@ -0,0 +1,13 @@
+# ✏️ Edit curr abbrs
```
```suggestion
# ✏️ Edit current abbreviations
```

kisvegabor commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,12 @@
+
+-  `global` • understood by everyone
+-  `significant` • seen for the first time you still know what it means
+
```
I think we should add more info here, but it's not blocking now.

T1xx1 replied:
Yeah, but for now I don't have other ideas. Maybe the "how to abbr" with letters or syllables.

kisvegabor commented:
Thank you, I added few comments. I think the reworked version look really great!

T1xx1 commented:
And there. These should be the last changes.

T1xx1 commented:
And there. These should be the last changes.

philipeachille commented:
> > "Meanwhile we wait for @philipeachille response" @T1xx1 what do you wish me to comment on?
> 
> Just to see if you agree with the changes.

I do follow the conversations in this repo. If I don't comment, it's like "agree"...

philipeachille commented:
> And there. These should be the last changes.

Congrats! Looking forward to the lat/lng fight!

kisvegabor commented:
LGTM! Thank you very much!

Please make it non-draft and I'll merge it.

kisvegabor commented:
I've found the button :smile:

T1xx1 commented:
> LGTM! Thank you very much!
> 
> Please make it non-draft and I'll merge it. 

Yes I was going to do it today.

T1xx1 commented:
![Screenshot_20230213_101221_com.github.android.jpg](https://user-images.githubusercontent.com/69322987/218417531-da35cf9f-7fe8-4236-afc2-2bbfc00e318e.jpg)

HAHAH, rip the format on gt.

---

BTW I've found an error

![Screenshot_20230213_101239_com.github.android.jpg](https://user-images.githubusercontent.com/69322987/218417730-9403c49d-e8b8-4d9d-927f-5032d45d5bc6.jpg)

Node has ° in the wrong position.
@kisvegabor could you fix it.

T1xx1 commented:
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository

Issue templates setup to make templates available as issues.

kisvegabor commented:
> Node has ° in the wrong position.

Fixed here: https://github.com/kisvegabor/abbreviations-in-code/commit/ff1d8819da6f863b217da2a252de1c23af0bfd0c

> Issue templates setup to make templates available as issues.

I've did it with the legacy method but it's working now.
